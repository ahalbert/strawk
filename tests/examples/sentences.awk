BEGIN {
  count=0
  words=0
}
# This regex matches each sentence in the input, and then prints the sentence. It works across newlines!
/(?s)(.*)\./ { 
  #gsub (Global SUBstitution replaces all newlines in the string with an empty string
  sentence = gsub(/\n/, "")
  sentence = sub(/^ /, "", sentence) #sub only replaces the first occurance of a regex
  print sentence
  splitwords = split(sentence, " ")
  count += 1
  words += length(splitwords)
}

END { print "Average length of sentence in text:", (words / count) }
