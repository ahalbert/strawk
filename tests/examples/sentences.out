Sunt molestias autem doloremque.
Sed ut doloremque occaecati quo est quam numquam exercitationem suscipit et.
Ad enim voluptatem consequatur vitae quis.
Maxime quasi magni velit eius nam aut esse voluptatibus quis velit repellendus.
Temporibus facilis ut porro deleniti excepturi quas alias placeat.
Numquam minus aut doloribus fugit magni dolorum.
Omnis ut minus rem quo est qui voluptate iste impedit.
Vel impedit qui qui sit explicabo assumenda recusandae voluptatem quia animi.
Average length of sentence in text: 8.75
